AppInspector.RulesEngine/RulesVerifier.cs

```diff
@@ -142,18 +142,27 @@ public RuleStatus CheckIntegrity(ConvertedOatRule convertedOatRule)
         {
             if (searchPattern.PatternType == PatternType.RegexWord || searchPattern.PatternType == PatternType.Regex)
             {
+                RegexOptions regexOpts = Utils.RegexModifierToRegexOptions(searchPattern.Modifiers, _options.EnableNonBacktrackingRegex);
                 try
                 {
                     if (string.IsNullOrEmpty(searchPattern.Pattern))
                     {
                         throw new ArgumentException();
                     }
-
-                    RegexOptions regexOpts =
-                        Utils.RegexModifierToRegexOptions(searchPattern.Modifiers ?? Array.Empty<string>());
-
+                                        
                     _ = new Regex(searchPattern.Pattern, regexOpts);
                 }
+                catch (NotSupportedException e)
+                {
+                    // Regex requires backtracking, try without non-backtracking
+                    _logger?.LogWarning(MsgHelp.GetString(MsgHelp.ID.VERIFY_RULES_REGEX_FAIL), rule.Id ?? "", searchPattern.Pattern ?? "", e.Message);
+#if NET7_0_OR_GREATER
+                    // remove non-backtracking flag by applying XOR
+                    regexOpts ^= RegexOptions.NonBacktracking;
```

Comment:
```suggestion
                    regexOpts &= ~RegexOptions.NonBacktracking;
```

I think its a bit clearer to just not the specific bit rather than xoring.

Reply:
Yeah, changed the logic, a bit of edge case, but your option is better.